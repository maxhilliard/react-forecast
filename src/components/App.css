@import url('https://fonts.googleapis.com/css?family=Roboto+Mono');
@value screen-sm from '../styles/variables.css';

* {
    font-family: 'Roboto Mono', monospace;
    text-align: center;
    color: white;
}

.container {
    width: 100vw;
    height: 100vh;
    padding: 5px;
    box-sizing: border-box;

    background-color: black;

    display: grid;
    grid-template-columns: 1fr;
    grid-template-rows: 3fr 1fr;
    grid-column-gap: 20px;
    grid-row-gap: 20px;
    grid-template-areas:
        "forecast"
        "search"
};

.forecastContainer {
    grid-area: forecast;
}

.searchContainer {
    grid-area: search;
}

@media (min-width: screen-sm) {
    .container {
        padding: 40px;
        grid-template-columns: 1fr 1fr;
        grid-template-rows: 1fr;
        grid-template-areas:
        "forecast search"
    }
}