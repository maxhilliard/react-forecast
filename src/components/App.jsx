import React, { Component } from 'react';

import { loadConfig } from '../config';

import Search from './search/Search';
import Forecast from './forecast/Forecast';

// TODO - use react-css-loader
import styles from './App.css';

const Loading = () => (
    <p>
        Loading
    </p>
);

class App extends Component {
    constructor(props) {
        super(props);

        this.state = {
            isConfigLoaded: false,
            forecast: [],
        };

        this.handleForecastResponse = this.handleForecastResponse.bind(this);
    }

    async componentDidMount() {
        await loadConfig();

        this.setState(prevState => ({ ...prevState, isConfigLoaded: true }));
    }

    handleForecastResponse(forecast) {
        this.setState(prevState => ({
            ...prevState,
            forecast,
        }));
    }

    render() {
        const { isConfigLoaded, forecast } = this.state;

        if (!isConfigLoaded) return <Loading />;

        return (
            <div className={styles.container}>
                <Forecast forecast={forecast} />
                <Search handleForecastResponse={this.handleForecastResponse} />
            </div>
        );
    }
}

export default App;
