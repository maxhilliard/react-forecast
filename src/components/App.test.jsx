import React from 'react';
import { shallow } from 'enzyme';

import App from './App';

describe('The app component', () => {
    it('should match snapshot', () => {
        const wrapper = shallow(<App />);
        wrapper.setState({ isConfigLoaded: true });

        expect(wrapper).toMatchSnapshot();
    });
});
