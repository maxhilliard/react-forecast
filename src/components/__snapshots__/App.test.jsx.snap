// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The app component should match snapshot 1`] = `
<div
  className="container"
>
  <Forecast
    forecast={Array []}
    isForecastError={false}
  />
  <Search
    handleErrorResponse={[Function]}
    handleForecastResponse={[Function]}
  />
</div>
`;
